src/js/components/SingleImage.vue

```diff
@@ -275,22 +255,34 @@ export default {
           this.artefact = 'new'
           this.artName = to.params.artID.split('name-')[1]
         } else {
-          this.artefact = to.params.artID
-          this.scrollVersionID = to.params.scrollVersionID
-          this.fetchArtefactMask()
+          this.artefact = to.params.artID >>> 0
+          this.scrollVersionID = to.params.scrollVersionID >>> 0
+          if(this.corpus.artefacts.get(this.artefact).mask === '') {
+            // this.$store.commit('addWorking')
+            this.corpus.artefacts.fetchMask(to.params.scrollVersionID, to.params.artID)
+            .then(res => {
+              // this.$store.commit('delWorking')
+              this.firstClipMask = this.clipMask = wktPolygonToSvg(this.corpus.artefacts.get(this.artefact).mask)
+            })
+          } else {
+            this.firstClipMask = this.clipMask = wktPolygonToSvg(this.corpus.artefacts.get(this.artefact).mask)
+          }
         }
         this.lock = false
       }
     }
   },
   filters: {
     formatImageType(value) {
-      if (!value) return ''
-      let formattedString = value.start === value.end ? value.start : value.start + '–' + value.end
-      if (value.type >>> 0 == 2) {
-        formattedString += ' RL'
-      } else if (value.type >>> 0 == 3) {
-        formattedString += ' RR'
+      let formattedString = ''
+      if (value == 0) {
```

Comment:
suggest a switch here.

Reply:
Good call.  I'm always allergic to switches, but it would be more clear here.